### ISSUE TITLE
Misleading error message on processing instruction - Xml(UnexpectedToken("in element", "'<?xml'", None))

### ISSUE THREAD

**Issue by Mingun** [horazont/rxml#11]
The parsing of https://github.com/Mingun/fast-xml/blob/master/tests/sample_rss.xml failed with:
```
Benchmarking low-level API/rxml: Warming up for 3.0000 sthread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Xml(UnexpectedToken("in element", "'<?xml'", None))', benches\bench.rs:85:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The code:
```rust
static SOURCE: &str = include_str!("../../tests/sample_rss.xml");

fn low_level_comparison(c: &mut Criterion) {
    let mut group = c.benchmark_group("low-level API");
    group.bench_function("rxml", |b| {
        use rxml::{EventRead, Lexer, PullDriver};
        use rxml::parser::{RawEvent, RawParser};

        type Parser<T> = PullDriver<T, RawParser>;

        b.iter(|| {
            let mut r = Parser::wrap(SOURCE.as_bytes(), Lexer::new(), RawParser::new());

            let mut count = criterion::black_box(0);
            loop {
                match r.read().unwrap() {
                    Some(RawEvent::ElementHeadOpen(..)) => count += 1,
                    None => break,
                    _ => (),
                }
            }
            assert_eq!(count, 1550, "Overall tag count in ./tests/sample_rss.xml");
        })
    });
}
```

[fast-xml](https://lib.rs/crates/fast-xml), [maybe_xml](https://lib.rs/crates/maybe_xml), [rapid-xml](https://lib.rs/crates/rapid-xml), [xmlparser](https://lib.rs/crates/xmlparser), [xml_oxide](https://lib.rs/crates/xml_oxide), [xml5ever](https://lib.rs/crates/xml5ever) and [xml_rs](https://lib.rs/crates/xml-rs) is able to parse it.

**Comment by horazont** [comment]
Thanks for the report.

Mind that the `r` in `rxml` is for "restricted", not for Rust. Among other things it explicitly rejects are processing instructions, which are contained on line 2 in your sample (`<?xml-stylesheet ...`).

I took the liberty of retitling the issue, because the only bug in rxml here is that it does not emit a `RestrictedXml` error for this case (instead of a confusing unexpected token).

**Comment by horazont** [comment]
I just released 0.8.1 with a fix for this issue. Please feel free to reopen or file a follow-up issue if it doesn't work as expected or you uncover more of these cases. Thanks!
